.github/workflows/release-temporal.yml

```diff
@@ -35,4 +35,4 @@ jobs:
           SRC_REPO: temporaliotest
           DST_REPO: temporalio
           LATEST: ${{ github.event.inputs.latest }}
-        run: go run src/release_temporal/main.go
+        run: go run src/release_temporal/main.go --commit "$COMMIT" --dst-tag "$TAG" --images "$IMAGES" --src-repo "$SRC_REPO" --dst-repo "$DST_REPO"
```

Comment:
Seems a bit weird that we're setting env vars to just supply them on a command line. Also, some of these are not being passed as arguments (just relying on the env var) and some are. What's the distinction?

Reply:
As far as I know, there's no way to do correct string interpolation in GHA without using environment variables like this.

If this was written like:
```
go run src/release_temporal/main.go --dst-tag "${{ github.event.inputs.tag }}" ...
```
and then the value of `tag` was `" ; cat /etc/passwd` then you'd get a shell injection. Please correct me if you know a better way

Reply:
You are correct about interpolation: https://securitylab.github.com/research/github-actions-untrusted-input/. Still not clear why some things are passed as arguments but others not.

Reply:
*My opinion*
There are three types of argument-passing strategies:
* config file - is the same across environments and doesn't often change - for example, defaults.
* environment variables - are different across environments but don't change across invocations - for example, CI and local passwords.
* command arguments - are different across invocations.

So I mostly follow this strategy: to pass values as arguments that differ across invocations, set values as environment variables that are the same across environments, and use config files that mostly don't change.

I see that arguments `LATEST`, `IMAGES`, `SRC_REPO`, and `DST_REPO` don't follow that strategy. I don't recall why I made this decision, and I can fix that by following my strategy. Probably because I wanted to log those arguments in the GHA build log. But I see that GHA provides env vars.
```
Run go run src/release_temporal/main.go
  go run src/release_temporal/main.go
  shell: /usr/bin/bash -e {0}
  env:
    COMMIT: 30fde9a1f1456c245151bce771f254a975ab0bb8
    TAG: 1.20.0.0
    USERNAME: ***
    PASSWORD: ***
    IMAGES: server auto-setup admin-tools
    SRC_REPO: temporaliotest
    DST_REPO: temporalio
    LATEST: true
```

However, the main change in this PR is to accept arguments as env vars or cli args. For example:
```
flagSet.StringVar(&args.commit, "commit", os.Getenv("COMMIT"), "Commit hash")
```